pkg/mark/link.go

```diff
@@ -0,0 +1,73 @@
+package mark
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/kovetskiy/mark/pkg/confluence"
+	"github.com/reconquest/pkg/log"
+)
+
+type Link struct {
+	MDLink string
+	Link   string
+}
+
+// ResolveRelativeLinks finds links in the markdown, and replaces one pointing
+// to other Markdowns either by own link (if not created to Confluence yet) or
+// witn actual Confluence link
+func ResolveRelativeLinks(
+	api *confluence.API,
+	markdown []byte,
+	base string,
+) ([]Link, error) {
+	links := []Link{}
+	re := regexp.MustCompile("\\[.*?\\]\\((.*?)\\)")
+	submatchall := re.FindAllStringSubmatch(string(markdown), -1)
+	for _, element := range submatchall {
+		possibleMDFile := element[1]
+		filepath := filepath.Join(base, possibleMDFile)
+		if _, err := os.Stat(filepath); err == nil {
+			linkMarkdown, err := ioutil.ReadFile(filepath)
+			if err != nil {
+				log.Errorf(err, "unable to read markdown file %s", filepath)
```

Comment:
This is much better now, but don't forget that the function has `([]Link, error)` returning signature, so we don't need to log errors here, we should return the first occurred error. 

```
if err != nil
    return nil, karma.Format(err, "unable to read markdown file: %s", filepath)
}
```

Reply:
Hmm... maybe then some list of failing ones should be made, as I don't think it's reasonable to end if first one finds error. So no return in the loop, but after it